### PR TITLE
Pointwise ABC, MonoT5, New Unit Tests

### PR DESCRIPTION
**Pull request by xpbowler**
## Summary of Changes

- Bug fix in `extract_kwargs` function 
- Bug fix where IdentityReranker not returning top_k_rerank (only top_k_retrieve)
- Add unit tests for API and `retrieve_and_rerank`
- Added pointwise and pairwise ABC
- Bug fix in `Rank_LLM` ABC parameters (missing kwargs)
...more changes below (@IR3KT4FUNZ)

### CODE REVIEW

**Comment by xpbowler** [comment]
@IR3KT4FUNZ

**Comment by IR3KT4FUNZ** [comment]
Changes: 
- add monot5 as a promptmode, and MonoT5 subclass of PointwiseRankLLM
- write implementations for the abstract functions for MonoT5 and PointwiseRankLLM classes
- update Reranker class to add support for the MonoT5 agent

File: test/api/output.txt
Comment (ronakice) [inline test/api/output.txt:1]: What is this?

File: src/rank_llm/rerank/pointwise/monot5.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                skip_special_tokens=True,
+                spaces_between_special_tokens=False,
+            )
+            for single_token_sequence in batch_output_ids
+        ]
+
+        for logit_tensor in batch_logits[0]:
+            truth_logit = logit_tensor[1176]
+            false_logit = logit_tensor[6136]
+            score = math.exp(truth_logit) / (
```
Comment (ronakice) [inline src/rank_llm/rerank/pointwise/monot5.py:75]: this is hardcoded and won't scale in the future to other T5 base models, note this as a T5 todo

File: src/rank_llm/rerank/pointwise/monot5.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        logits = output.scores
+
+        if self._llm.config.is_encoder_decoder:
+            output_ids = output_ids[0]
+            output_ids = output_ids[1:]
+
+        outputs = self._tokenizer.decode(
+            output_ids, skip_special_tokens=True, spaces_between_special_tokens=False
+        )
+        truth_logit = logits[0][0][1176]
```
Comment (ronakice) [inline src/rank_llm/rerank/pointwise/monot5.py:107]: same here

**Review by ronakice — APPROVED** [review]
LGTM!

**Pull request merged**